crate/agent/src/service.rs

```diff
@@ -1,49 +1,121 @@
 use crate::error::Error;
+use serde::Deserialize;
+use std::process::Command;
+use sysinfo::{ProcessExt, System, SystemExt};
 
-pub(crate) mod internal {
-    use std::process::Command;
-
-    use crate::error::Error;
-
-    #[doc(hidden)]
-    pub trait UnixServiceInternal {
-        const NAME: &'static str;
-
-        fn call(actions: &[&str]) -> Result<String, Error> {
-            match Command::new(Self::NAME).args(actions).output() {
-                Ok(output) => {
-                    if output.status.success() {
-                        Ok(String::from_utf8_lossy(&output.stdout).to_string())
-                    } else {
-                        Err(Error::Command(format!(
-                            "Output: {} - error: {}",
-                            String::from_utf8_lossy(&output.stdout),
-                            String::from_utf8_lossy(&output.stderr)
-                        )))
-                    }
-                }
-                Err(e) => Err(Error::Command(e.to_string())),
+pub(crate) fn _call(exe: &str, args: &[&str]) -> Result<String, Error> {
```

Comment:
no need to make it `pub(crate)`
```suggestion
fn _call(exe: &str, args: &[&str]) -> Result<String, Error> {
```